next/src/modules/homeChart/components/ZoomView.tsx

```diff
@@ -32,12 +39,54 @@ const ZoomView: React.FC<Props> = ({
   } = useQuery<EcgFragment, Error>(['record', id], () => getFragment(id));
 
   const handleDecision = (choice: Choice) => {
+    if (decision?.choice === choice) {
+      return;
+    }
+
+    if (decision) {
+      setSelectedDecision(choice);
+      setIsConfirmModal(true);
+
+      return;
+    }
+
+    if (!id) {
+      return;
+    }
+
     addFeedback(id, choice);
     onClose();
   };
 
+  const handleConfirm = () => {
+    if (!decision || !selectedDecision) {
+      return;
+    }
+
+    if (!decision.id) {
+      return;
+    }
+
+    addFeedback(decision.id, selectedDecision);
+    setIsConfirmModal(false);
+  };
+
+  const handleCancel = () => {
+    setIsConfirmModal(false);
+  };
+
   return (
     <>
+      <Modal
+        title="Confirmation"
+        centered
+        visible={isConfirmModal}
+        onOk={handleConfirm}
+        onCancel={handleCancel}
+      >
+        <p>Are you sure you want to change the feedback?</p>
```

Comment:
```suggestion
        <p>Are you sure you want to change the label?</p>
```

This question makes more sense that way